pages/advanced/api-service-architecture/safe-transaction-service/interacting-with-api.md

```diff
@@ -0,0 +1,576 @@
+# Interacting with API
+
+This section shows some examples of how to interact with the Safe Transaction Service API in different situations.
+
+The examples have been proposed using [Curl](https://github.com/curl/curl) requests, the [Safe Apps SDK](https://github.com/safe-global/safe-apps-sdk) and the [safe-eth-py](https://github.com/safe-global/safe-eth-py) python library.
+
+
+## Prerequisites
+
+1. [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm#using-a-node-version-manager-to-install-nodejs-and-npm).
+2. [Python](https://www.python.org/downloads/) >= 3.9 (Python 3.12 is recommended).
+3. Have a Safe configured with threshold equal to 2, where 2 signatures are needed to execute transactions.
+
+## Install dependencies
+
+```bash
+yarn add ethers @safe-global/api-kit @safe-global/protocol-kit @safe-global/safe-core-sdk-types
+```
+
+```bash
+pip install safe-eth-py web3 hexbytes
+```
+
+## Examples
+
+### Creating a transaction and sending offchain signatures from the owners
+
+#### Using Curl
+
+- Create transaction
+
+```bash
+    curl -X 'POST' \
+    'https://safe-transaction-sepolia.safe.global/api/v1/safes/0xc62C5cbB964ffffffffff82f78A4d30713174b2E/multisig-transactions/' \
+    -H 'accept: application/json' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "safe": "0xc62C5cbB964459F3C984682f78A4d3ffffffffff",
+        "to": "0x795D6C88B4Ea3CCffffffffffCa8a11Bc0496228",
+        "value": 2000000000000000,
+        "data": null,
+        "operation": 0,
+        "gasToken": "0x0000000000000000000000000000000000000000",
+        "safeTxGas": 0,
+        "baseGas": 0,
+        "gasPrice": 0,
+        "refundReceiver": "0x0000000000000000000000000000000000000000",
+        "nonce": 15,
+        "contractTransactionHash": "0x56b2931d1053b6afffffffffff3ba29b5c2baafdf1a588850da72a62674941b6",
+        "sender": "0xAA86E576c084aCFa56fc4D0E17967ffffffffff8",
+        "signature": "0x6a2b57023af16241511619ea95f7cd03d00aa6b79d1ca80e21a0b89cd2c38ffffffffff9b738ffbd680c4d717b9b0c9eae568f3edebc40a0c004700bffffffffff"
+    }'
+```
+
+- Add signatures to transaction
+
+```bash
+    curl -X 'POST' \
+    'https://safe-transaction-sepolia.safe.global/api/v1/multisig-transactions/0x56b2931ffffffffff303bbaadf3ba29b5c2baafdf1a5ffffffffff62674941b6/confirmations/' \
+    -H 'accept: application/json' \
+    -H 'Content-Type: application/json' \
+    -d '{
+        "signature": "0x8c41aaa029db2942a1574c3ec9442b80764e953f21f994edb641d33ffffffffffa55ffffffffffefcaea5b3360bd4c789803fe289b2ad467fc9b3bedb479dad20"
+    }'
+```
+
+#### Using safe-core-sdk
+
+```js
+    async function createTransactionAndSignByOwners() {    
```

Comment:
I would suggest that you add the imports of the used libs so they are easier to understand
```suggestion
```js
    import { ethers } from 'ethers'
    import SafeApiKit from '@safe-global/api-kit'
    import Safe, { EthersAdapter } from '@safe-global/protocol-kit'
    import { MetaTransactionData } from '@safe-global/safe-core-sdk-types'

    async function createTransactionAndSignByOwners() {    
```